### DISCUSSION TITLE
Create a separate user manual using the reflection capabilities of Markdown

### DISCUSSION THREAD

**Discussion by Witiko** [Witiko/markdown#135] [Ideas]
The Markdown package is written as a single huge documented TeX (DTX) file combining code and documentation. This keeps the documentation close to the code, but it also makes it difficult for contributors to orient themselves in the code and to produce documentation that is easy to read for users and whose flow is not dictated by the structure of the code. Thanks for @writersglen for hammering this point home.

In #119 and #124, we have added extensive reflection capabilities that allow us to enumerate and inspect high-level concepts such as options, renderers, and renderer prototypes. The motivation for this was the reduction of technical debt and to make it easier to automatically construct high-level interfaces for new TeX formats (e.g. #17). However, if a free-text description and examples for the high-level concepts could also be retrieved via reflection, then we could write a separate document as a user manual and automatically generate lists of options, renderers, and renderer prototypes using reflection.

Separating the user manual into an independent document would allow us to shrink the DTX file and allow for separation of labor, where the user manual is written from the viewpoint of a user and independent on the structure of the code.

**Comment by Witiko** [comment]
@writersglen In fact, we could also use reflection to automatically generate some of the appendices in Publish Beautiful.

**Comment by writersglen** [comment]
Reflection? Sounds interesting.

Best,

L.

**Comment by writersglen** [comment]
Ah, I see. Very interesting. Way to go.

Best,

L.
